src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCell.cs

```diff
@@ -966,43 +972,44 @@ protected override void OnKeyDown(KeyEventArgs e)
         {
             if (!e.Handled)
             {
-                const ModifierKeys ModifierMask = ModifierKeys.Alt | ModifierKeys.Control | ModifierKeys.Shift | ModifierKeys.Windows;
-                ModifierKeys modifierKeys = Keyboard.Modifiers & ModifierMask;
-
-                if (((e.SystemKey == Key.Right) || (e.SystemKey == Key.Left)) && (modifierKeys == ModifierKeys.Alt))
+                if (!OptOutOfGridColumnResizeUsingKeyboard)
                 {
-                    DataGridLength updatedWidth = new DataGridLength();
-
-                    if (e.SystemKey == Key.Right)
+                    ModifierKeys modifierKeys = Keyboard.Modifiers & ModifierMask;
+                    if (((e.SystemKey == Key.Right) || (e.SystemKey == Key.Left)) && (modifierKeys == ModifierKeys.Alt))
                     {
-                        updatedWidth = new DataGridLength(Column.ActualWidth + ColumnWidthStepSize);
-                    }
-                    else if (e.SystemKey == Key.Left)
-                    {
-                        updatedWidth = new DataGridLength(Column.ActualWidth - ColumnWidthStepSize);
-                    }
+                        DataGridLength updatedWidth = new DataGridLength();
 
-                    if (Column != null)
-                    {
-                        if (Column.CanColumnResize(updatedWidth))
+                        if (e.SystemKey == Key.Right)
                         {
-                            Column.SetCurrentValueInternal(DataGridColumn.WidthProperty, updatedWidth);
+                            updatedWidth = new DataGridLength(Column.ActualWidth + ColumnWidthStepSize);
                         }
-                        e.Handled = true;
+                        else if (e.SystemKey == Key.Left)
```

Comment:
```suggestion
                        else
```

No need for the `else if` because the check has already been done in the outer if.

Reply:
Made the changes as suggested